### PR TITLE
KOGITO-10058: Migrate serverless-workflow-diagram-editor from private to KIE repos and libs

### PR DESCRIPTION
**Pull request by treblereel**
jira: https://issues.redhat.com/browse/KOGITO-10058

### CODE REVIEW

**Comment by LuboTerifaj** [comment]
Hello @treblereel , could you please provide the details about the changes in related issue/jira? Thank you!

**Review by LuboTerifaj — CHANGES_REQUESTED** [review]
Hello @treblereel ,

`org.treblereel.*` dependencies were not removed from YARD editor.

Please see also my [KOGITO-10058-comment](https://issues.redhat.com/browse/KOGITO-10058?focusedId=24339208&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-24339208)

Other than that, all integration tests passed  and I didn't find any issues.

Thank you!

**Comment by treblereel** [comment]
> Hello @treblereel ,
> 
> `org.treblereel.*` dependencies were not removed from YARD editor.
> 
> Please see also my [KOGITO-10058-comment](https://issues.redhat.com/browse/KOGITO-10058?focusedId=24339208&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-24339208)
> 
> Other than that, all integration tests passed and I didn't find any issues.
> 
> Thank you!

I think we should ask Eder and Mark to clarify this, but as far as I remember from our discussion, it's because of the Apache license: we can't move to Apache projects created by third parties.

Yard is @Rikkola 's project, so I think we can ask him to update the deps. I have informed him already that we published libs to jboss/sonatype so maybe we can just wait for him.

**Comment by LuboTerifaj** [comment]
Hello @treblereel , [KOGITO-10058](https://issues.redhat.com/browse/KOGITO-10058) scope is to remove all dependencies from kie-tools repository. If you want to split the work with @Rikkola , we can do that, but we need to update the scope of the jira and report remaining work in a separate issue. Do you want to split the work?

**Comment by LuboTerifaj** [comment]
One more note to my previous comment.

Alternatively, we can keep the scope of the jira the same, and link multiple PRs there, or use more commits in this PR and involve Toni here.

**Comment by Rikkola** [comment]
The model from YARD editor is going to move to incubator-kie-drools after 10.0.0. During the move I was planning to swap the treblereel yaml mapper dependency. If this needs to be done before 10.0.0, then I can swap it now.

**Review by handreyrc — APPROVED** [review]
Hey @treblereel,

It works well and I couldn't find any issues.  

Thanks for your work here!

**Comment by LuboTerifaj** [comment]
@Rikkola, @treblereel , how do you want to address the dependencies not removed from YARD editor? See my previous comments. Thanks!

**Comment by LuboTerifaj** [comment]
@Rikkola I think it should be done before 10.0.0 release, so there are no third party libraries references in the repository.

**Comment by Rikkola** [comment]
@LuboTerifaj @treblereel I can do the change. There will be some manual testing needed, so I will be the fastest one to do it.

**Comment by treblereel** [comment]
@Rikkola thank you, all green !

**Comment by Rikkola** [comment]
np

File: packages/yard-validator-worker/deps.txt
Comment (LuboTerifaj) [inline packages/yard-validator-worker/deps.txt:1]: @treblereel , @Rikkola , can you please remove this file?

**Review by LuboTerifaj — CHANGES_REQUESTED** [review]
Please see my inline comment. Thanks!

File: packages/yard-validator-worker/deps.txt
Comment (Rikkola) [reply packages/yard-validator-worker/deps.txt:1]: True, it should be removed.

**Review by LuboTerifaj — APPROVED** [review]
Thanks for the updates @treblereel , @Rikkola . It looks good to me now.

**Review by Rikkola — APPROVED** [review]

**Pull request merged**